src/store/reducers.js

```diff
@@ -0,0 +1,55 @@
+const defaultState = {
+  trendingGifs: [],
+  favoriteGifs: [],
+  error: '',
+  loading: false,
+  singleGif: null
+}
+
+const reducers = (state = defaultState, action) => {
+  switch(action.type) {
+    case 'LOADING_START':
+      return { ...state, loading: true }
+    case 'LOADING_END':
+      return { ...state, loading: false }
+
+    case 'FETCH_SUCCEEDED':
+      const trendingGifs = action.trendingGifs.data
+      return { ...state, trendingGifs };
+
+    case 'FETCH_FAILED':
+      return { ...state, error: action.payload.message }
+    case 'FETCHED_SINGLE':
+      return { ...state, singleGif: action.singleGif.data };
+    case 'FETCHED_SEARCH':
+      return { ...state, trendingGifs: action.searchGifs.data };
+
+    case 'FETCH_FAVORITES':
+      let fetchedGifsFromStorage = JSON.parse(localStorage.getItem("favorites"))
```

Comment:
Remember that reducers must be pure, the calls to `localStorage` are side effects.